src/Compilers/CSharp/Test/Emit/Emit/EntryPointTests.cs

```diff
@@ -107,6 +107,32 @@ public class D
                 Diagnostic(ErrorCode.ERR_MultipleEntryPoints, "Main"));
         }
 
+        [Fact]
+        public void ERR_MultipleEntryPoints_SyncAndAsync()
+        {
+            string source = @"
+using System.Threading.Tasks;
+
+public class C
+{
+  public static async Task Main() { await Task.Delay(1); }
+  public static void Main(string[] a) { System.Console.WriteLine(2); }
+}
+
+public class D
+{
+  public static string Main() { System.Console.WriteLine(3); return null; }
```

Comment:
Please add a test without this version.

Reply:
@333fred I'll add it once the current test pass.

Reply:
I don't see a test without this unrelated `Main`